src/hooks/useGenerator.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-restricted-syntax */
+import { useState } from 'react';
+
+/**
+ * 배열 형태의 generator를 받아,
+ * Generator 타입의 상태 generator와 reset 할 수 있는 함수를 반환합니다.
+ */
+const useGenerator = (
+  param: unknown[],
+): [generator: Generator, reset: () => void] => {
+  function* gen() {
+    for (const val of param) {
+      yield val;
+    }
+    // const yieldValue = (val: unknown) => yield val;
+    // param.forEach((p) => yieldValue(p));
+  }
```

Comment:
```suggestion
  function* gen() {
    for (let i = 0; i < param.length; i += 1) {
      yield param[i];
    }
  }
```

요렇게 하면 린트 에러가 안 나네요 !! 아래 블로그 참고했습니당 (https://infoscis.github.io/2018/01/31/ecmascript-6-iterators-and-generators/)
![image](https://user-images.githubusercontent.com/22045163/141326621-875aa397-0909-4c48-9f7a-4ffc5ffe9cc4.png)

Reply:
대박... 감사합니다!!